File: Hole.java
public class Hole extends Entity
{
   private double radius;


   public Hole(double x, double y, double r)
   {
      super(0, x, y, 0, 0, 0, 0, "images/8ball.png");
      radius = r;
   }

   public double getRadius()
   {
      return radius;
   }

   public boolean checkCollisions(Entity b)
   {
      if(b instanceof Ball)
      {
      
      
         return (this.getX() - b.getX()) * (this.getX() - b.getX()) + (this.getY() - b.getY()) * (this.getY() - b.getY()) < radius * radius;
      
      
      }
      return false;
   }
   
   public void impact(Entity b)
   {
   if(b instanceof Ball)
   {
   ((Ball)b).setRadius(11);
   }
   
   
   }
}

File: Hole.kt
class Hole(x: Double, y: Double, val radius: Double) : Entity(0.0, x, y, 0.0, 0.0, 0.0, 0.0, "images/8ball.png") {

    override fun checkCollisions(b: Entity?): Boolean {
        return if (b is Ball) {
            (x - b.x) * (x - b.x) + (y - b.y) * (y - b.y) < radius * radius
        } else false
    }

    override fun impact(b: Entity?) {
        if (b is Ball) {
            b.radius = 11.0
        }
    }
}